doc/examples/segmentation/plot_quickshift_full_tree.py

```diff
@@ -0,0 +1,171 @@
+                                            parents_display.shape)])
+        ims.append(parents_display)
+        nb_levels = nb_levels + 1
+    nb_levels = nb_levels - 1
+    return ims[:-1], nb_levels
+
+ims, nb_levels = multilevel_segment(parents)
+ims_partial, nb_levels_partial = multilevel_segment(parents_partial)
+
+plt.set_cmap('viridis')
+fig, ax = plt.subplots(max(nb_levels, nb_levels_partial), 2, figsize=(20,
+                       max(nb_levels, nb_levels_partial) * 10),
+                       sharex=True, sharey=True)
+fig.suptitle('Segment the image based on the tree hierachy levels. \nThe left '
+             'column uses fullsearch=True while the right column does not.',
+             fontsize=12)
+for i in range(nb_levels):
+    ax[i, 0].imshow(ims[i])
+    ax[i, 0].set_title("full search level " + str(i))
+for i in range(nb_levels_partial):
+    ax[i, 1].imshow(ims_partial[i])
+    ax[i, 1].set_title("partial search level " + str(i))
+
+for a in ax.ravel():
+    a.set_axis_off()
+
+plt.tight_layout()
+plt.show()
+
+
+# perform multiscale segmentation i.e. segment based on the distance to parent
+def multiscale_segment(segments_quick, nb_scales, min_scale):
+    ims = []
+    last_scale = 0
+    for i in range(min_scale, nb_scales + min_scale, 1):
+        max_dist = 2**i
+        # remove parents with distance > max_dist
+        parent_flat = np.array(segments_quick[1]).astype('uint32').ravel()
+        old = np.zeros(parent_flat.shape).astype('uint32')
+        too_far = np.array(segments_quick[2]).ravel() > max_dist
+        parent_flat[too_far] = np.arange(parent_flat.shape[0])[too_far]
+        while np.any(old != parent_flat):
+            old = parent_flat
+            parent_flat = parent_flat[parent_flat]
+        ims.append(np.reshape(parent_flat, segments_quick[0].shape))
+        if len(ims) >= 2:
+            if np.any(ims[-1] != ims[last_scale-1]) or last_scale == 0:
+                last_scale = len(ims)
+    return ims[:last_scale]
+nb_scales = 8
+min_scale = 2
+ims = multiscale_segment(segments_quick, nb_scales, min_scale)
+ims_partial = multiscale_segment(segments_quick_partial, nb_scales, min_scale)
+
+plt.set_cmap('viridis')
+fig, ax = plt.subplots(max(len(ims), len(ims_partial)), 2,
+                       figsize=(20, max(len(ims), len(ims_partial)) * 10),
+                       sharex=True, sharey=True)
+fig.suptitle('Segment the image based on the distance to parent (scale). \n'
+             'The left column uses fullsearch=True while the right column '
+             'does not.', fontsize=12)
+for i in range(len(ims)):
+    ax[i, 0].imshow(ims[i])
+    ax[i, 0].set_title("full search scale " + str(2**(min_scale+i)))
+for i in range(len(ims_partial)):
+    ax[i, 1].imshow(ims_partial[i])
+    ax[i, 1].set_title("partial search scale " + str(2**(min_scale+i)))
+
+for a in ax.ravel():
+    a.set_axis_off()
+
+plt.tight_layout()
+plt.show()
```

Comment:
pls, add the output figure to this image :)

Reply:
Hello, thanks for the review :-)
I am not sure that I understand this properly. You want me to save the output as a png (or some other file format) and display it?

Reply:
I think borda asked for posting the figure on github.

Reply:
Ah! Ok. Here are all the figures. Left with full_search=True, and right with full_search=False. Output from quickshift
![ouput](https://user-images.githubusercontent.com/53205950/66184759-67635200-e6a7-11e9-9411-2e49983e1933.png)
Tree hierarchy segmentation (bottom to top)
![level](https://user-images.githubusercontent.com/53205950/66184766-69c5ac00-e6a7-11e9-98fe-82886126a411.png)
Usual distance based superpixel segmentation
![scale](https://user-images.githubusercontent.com/53205950/66184772-6b8f6f80-e6a7-11e9-82f8-bfb216a62ec7.png)